### PR TITLE
Show aggregate counts of roles for each Person

### LINKED ISSUE #799: Show how often taught on instructor page
Results of this report https://amy.software-carpentry.org/workshops/instructors_by_date
show often each instructor has taught.

This number should also display at bottom of Task list on each instructor profile page.

### PR DESCRIPTION
**Pull request by narayanaditya95**
This fixes #799 

Please review. Should I make changes to the way the counts are presented or this works?

### CODE REVIEW

**Comment by gvwilson** [comment]
Looks good to me - @pbanaszkiewicz please merge if you like it. @maneesha please let us know if you like the look.

File: workshops/templates/workshops/person.html [hunk 1]
```diff
@@ -73,6 +73,11 @@
       </li>
       {% endfor %}
     </ul>
+    <ul>
+      <li> Has taught {{ person.task_set.instructors.count }} time{{ person.task_set.instructors.count|pluralize }}</li>
+      <li> Has helped {{ person.task_set.helpers.count }} time{{ person.task_set.helpers.count|pluralize }}</li>
+      <li> Has learned {{ person.task_set.learners.count }} time{{ person.task_set.learners.count|pluralize }}</li>
```
Comment (pbanaszkiewicz) [inline workshops/templates/workshops/person.html]: This code yields 6 additional queries; please use annotations as I pointed in #799.

Comment (narayanaditya95) [inline workshops/templates/workshops/person.html]: However, annotations are defined on querysets.
Do you want me to write a query with the `pk` parameter?

Comment (pbanaszkiewicz) [inline workshops/templates/workshops/person.html]: You can use:
```
person = Person.objects.filter().annotate().filter(pk=person_id).first()
```

**Comment by narayanaditya95** [comment]
@pbanaszkiewicz Changes committed.

File: workshops/views.py [hunk 2]
```diff
@@ -470,7 +470,30 @@ def all_persons(request):
...
+        ),
+        num_learner=Count(
+            Case(
+                When(task__role__name='learner', then=Value(1)),
+                 output_field=IntegerField()
+            )
+        )
+    ).first()
+    if not person:
+        raise Http404
```
Comment (pbanaszkiewicz) [inline workshops/views.py]: Please change this to:

```python
try:
    person = Person.objects.annotate(annotations_here).get(id=person_id)
except Person.DoesNotExist:
    raise Http404('Person matching query does not exist.')
```

I just tested a very simplified annotation and `.get` and it worked, but ensure that these complex annotations work as well.

Comment (narayanaditya95) [inline workshops/views.py]: Also sheds about 25ms. 2.89ms -> 2.63 ms. +1

**Comment by pbanaszkiewicz** [comment]
Thanks, @narayanaditya95!

**Pull request merged**